waddrmgr/address.go

```diff
@@ -177,11 +177,12 @@ func (a *managedAddress) lock() {
 	a.privKeyMutex.Unlock()
 }
 
-// Account returns the account number the address is associated with.
+// InternalAccount returns the internal account number the address is associated
+// with.
 //
 // This is part of the ManagedAddress interface implementation.
-func (a *managedAddress) Account() uint32 {
-	return a.derivationPath.Account
+func (a *managedAddress) InternalAccount() uint32 {
+	return a.derivationPath.InternalAccount
```

Comment:
Correct. The distinction was necessary for imported accounts, as they could have a 0 account number, but internally they'd be mapped to a 1 if a default account already existed before the import. This would propagate all the way down to the derivation path that gets populated in the PSBTs, resulting in a signature for the wrong key.